opennms-webapp/src/test/java/org/opennms/web/event/EventControllerIT.java

```diff
@@ -0,0 +1,259 @@
+        alarmQueryParms.filters = filterList;
+        alarmQueryParms.limit = 20;
+        alarmQueryParms.sortStyle = org.opennms.web.alarm.SortStyle.ID;
+        alarmQueryParms.display = "Y";
+        alarmQueryParms.multiple = 0;
+
+        AlarmCriteria alarmCriteria = new AlarmCriteria(alarmQueryParms);
+
+        when(m_webAlarmRepository.getMatchingAlarms(any())).thenReturn(getAlarms());
+        when(m_webAlarmRepository.countMatchingAlarms(AlarmUtil.getOnmsCriteria(alarmCriteria))).thenReturn(1);
+
+        ModelAndView mv = alarmFilterController.list(request,response);
+        OnmsAlarm[] av = (OnmsAlarm[])mv.getModelMap().get("alarms");
+        Assert.assertEquals(1, av.length);
+        Assert.assertEquals("Normal", av[0].getSeverity().getLabel());
+    }
+
+    private List<OnmsEvent> getOnmsEvents(){
+        OnmsEvent event1 = new OnmsEvent();
+        event1.setId(1);
+        event1.setEventCreateTime(new Date());
+        event1.setEventDescr("test");
+        event1.setEventHost("localhost");
+        event1.setEventLog("Y");
+        event1.setEventDisplay("Y");
+        event1.setEventLogGroup("event dao test log group");
+        event1.setEventLogMsg("test");
+        event1.setEventSeverity(OnmsSeverity.CRITICAL.getId());
+        event1.setEventSource("EventDaoTest1");
+        event1.setEventTime(new Date());
+        event1.setEventUei("uei://org/opennms/test/EventDaoTest1");
+        event1.setEventParameters(Lists.newArrayList(
+                new OnmsEventParameter(event1, "user", "test", "string"),
+                new OnmsEventParameter(event1, "ds", "(memAvailReal + memCached) / memTotalReal * 100.0", "string"),
+                new OnmsEventParameter(event1, "description", "test", "string"),
+                new OnmsEventParameter(event1, "value", "4.7", "string"),
+                new OnmsEventParameter(event1, "instance", "node1", "string"),
+                new OnmsEventParameter(event1, "instanceLabel", "node1", "string"),
+                new OnmsEventParameter(event1, "resourceId", "node[70].nodeSnmp[]", "string"),
+                new OnmsEventParameter(event1, "threshold", "5.0", "string"),
+                new OnmsEventParameter(event1, "trigger", "2", "string"),
+                new OnmsEventParameter(event1, "rearm", "10.0", "string")));
+
+
+        List<OnmsEvent> onmsEventList = new ArrayList<>();
+        onmsEventList.add(event1);
+        return onmsEventList;
+    }
+
+    private OnmsAlarm[] getAlarms() {
+        OnmsAlarm alarm = new OnmsAlarm();
+        alarm.setUei("uei.opennms.org/vendor/Juniper/traps/jnxVpnIfUp");
+        alarm.setLastEvent(getOnmsEvents().get(0));
+        alarm.setSeverityId(3);
+        alarm.setCounter(100);
+        alarm.setLastEvent(getOnmsEvents().get(0));
+        alarm.setLogMsg("test");
+        OnmsAlarm[] alarms = {alarm};
+        return alarms;
+    }
+
+
+    private Event mapOnmsEventToEvent(OnmsEvent onmsEvent){
```

Comment:
Do you need this mapping? 
Can you directly create Event

Reply:
Created events directly now